internal/cephfs/nodeserver.go

```diff
@@ -127,13 +128,47 @@ func maybeUnlockFileEncryption(
 	stagingTargetPath string,
 	volID fsutil.VolumeID,
 ) error {
-	if volOptions.IsEncrypted() {
-		log.DebugLog(ctx, "cephfs: unlocking fscrypt on volume %q path %s", volID, stagingTargetPath)
 
+	if volOptions.IsEncrypted() == false {
+		return nil
+	}
+
+	// Define Mutex Lock variables
+	lockName := string(volID) + "-mutexLock"
+	lockCookie := lockName + "-coockie"
+	lockDesc := "Lock for " + string(volID)
+	lockDuration := 40 * time.Second
+	var flags byte = 0
+
+	log.DebugLog(ctx, "Creating lock for the following volume ID %s", volID)
+
+	ioctx, err := volOptions.GetConnection().GetIoctx(volOptions.MetadataPool)
+	if err != nil {
```

Comment:
return error from here?

Reply:
Hey. Thank you for the review. I have addressed most of your comments.

Reply:
if creation of the ioctx fails, an error should be returned

Reply:
@Sunnatillo, this is the last error that still needs to be handled. If getting the ioctx fails, there is no need to continue.